v3/generator/operations/operations.go

```diff
@@ -0,0 +1,694 @@
+package operations
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"os"
+	"strings"
+	"text/template"
+
+	"github.com/exoscale/egoscale/v3/generator/helpers"
+	"github.com/exoscale/egoscale/v3/generator/schemas"
+	"github.com/pb33f/libopenapi"
+	"github.com/pb33f/libopenapi/datamodel/high/base"
+	v3 "github.com/pb33f/libopenapi/datamodel/high/v3"
+)
+
+func Generate(doc libopenapi.Document, path, packageName string) error {
+	model, errs := doc.BuildV3Model()
+	for _, err := range errs {
+		if err != nil {
+			return fmt.Errorf("errors %v", errs)
+		}
+	}
+
+	output := bytes.NewBuffer(helpers.Header(packageName, "v0.0.1"))
+	output.WriteString(fmt.Sprintf(`package %s
+	
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"context"
+	"net/url"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/go-playground/validator/v10"
+)
+`, packageName))
+
+	methods := []Method{}
+
+	err := helpers.ForEachMapSorted(model.Model.Paths.PathItems, func(path string, item any) error {
+		pathItems := item.(*v3.PathItem)
+		return helpers.ForEachMapSorted(pathItems.GetOperations(), func(opName string, op any) error {
+			operation := op.(*v3.Operation)
+			funcName := helpers.ToCamel(operation.OperationId)
+			if funcName == "" {
+				funcName = helpers.ToCamel(path)
+			}
+
+			schemaResponses, err := renderResponseSchema(funcName, operation)
+			if err != nil {
+				return err
+			}
+			output.Write(schemaResponses)
+
+			schemaRequest, err := renderRequestSchema(funcName, operation)
+			if err != nil {
+				return err
+			}
+			output.Write(schemaRequest)
+
+			schemaParameters, err := renderMethodParameterSchema(funcName, operation)
+			if err != nil {
+				return err
+			}
+			output.Write(schemaParameters)
+
+			method, err := serializeMethod(path, opName, funcName, operation)
+			if err != nil {
+				return err
+			}
+
+			methods = append(methods, *method)
+
+			m, err := renderMethod(method)
+			if err != nil {
+				return err
+			}
+			output.Write(m)
+
+			return nil
+		})
+	})
+	if err != nil {
+		return err
+	}
+
+	// Add extra methods here from static generated (operationUtils).
+	// TODO put this static list somewhere.
+	methods = append(methods, Method{Name: "Wait", Params: "context.Context, *Operation, ...OperationState", ValueReturn: "(*Operation, error)"})
+	methods = append(methods, Method{Name: "WithZone", Params: "APIZone", ValueReturn: "Client"})
+	methods = append(methods, Method{Name: "WithContext", Params: "context.Context", ValueReturn: "Client"})
+	methods = append(methods, Method{Name: "WithHttpClient", Params: " *http.Client", ValueReturn: "Client"})
+	methods = append(methods, Method{Name: "WithRequestMiddleware", Params: "RequestMiddlewareFn", ValueReturn: "Client"})
+
+	iface, err := renderInterface(methods)
```

Comment:
What's the objective in generating an interface with hundreds of methods ?

Reply:
It's for the mocking purpose, with no interface you can not mock the API calls...etc